src/crypto/index.ts

```diff
@@ -3221,14 +3221,15 @@ export class Crypto extends TypedEventEmitter<CryptoEvent, CryptoEventHandlerMap
         });
     }
 
-    public preprocessOneTimeKeyCounts(oneTimeKeysCounts: Map<string, number>): Promise<void> {
-        const currentCount = oneTimeKeysCounts.get("signed_curve25519") || 0;
-        this.updateOneTimeKeyCount(currentCount);
-        return Promise.resolve();
-    }
+    public processKeyCounts(oneTimeKeysCounts?: Record<string, number>, unusedFallbackKeys?: string[]): Promise<void> {
+        if (oneTimeKeysCounts) {
+            this.updateOneTimeKeyCount(oneTimeKeysCounts["signed_curve25519"]);
+        }
+
+        if (unusedFallbackKeys) {
+            this.setNeedsNewFallback(!unusedFallbackKeys.includes("signed_curve25519"));
+        }
```

Comment:
I think we should inline both `updateOneTimeKeyCount` and `setNeedsNewFallback`. They are both very short.

Reply:
Why ?

I think the code is clear and easy understable

Reply:
1: it means I have to go and look up the implementation of `updateOneTimeKeyCount` and `setNeedsNewFallback`, which is harder than just seeing the implementation inline.
2: it complicates the API of `Crypto`: we should not expose methods that we do not intend other classes to use. Though we could also fix that by making them `private`.

Reply:
I added comments about and it's futur inlining after the change in the sliding sync